loadtest/user/userentity/actions.go

```diff
@@ -464,6 +480,21 @@ func (ue *UserEntity) GetTeam(teamId string) error {
 	return ue.store.SetTeam(team)
 }
 
+func (ue *UserEntity) UpdateTeam(team *model.Team) error {
+	ok, err := ue.IsTeamAdmin()
+	if err != nil {
+		return err
+	}
+	if !ok {
+		return nil
+	}
```

Comment:
Same as above.

Reply:
The thing is that any user is eligible to call UpdateTeam, but then they are not aware whether the action will succeed or not. So it's kinda not the caller's fault that they don't have the permissions. We could return an error, but then the logs will be unnecessarily polluted with this. Essentially, the load tester is not getting any extra information by knowing that some user doesn't have UpdateTeam permissions.

That's my thinking behind it. If we want to return an error, then we don't even need this check. `UpdateTeam` will automatically fail if the user doesn't have permissions. Introducing `IsTeamAdmin` was the idea behind this.

Reply:
I agree on verbosity and there is a clear trade-off here, I'm just pointing out that we are considering this action as successful even if don't update. And we are logging and saying team updated [here](https://github.com/mattermost/mattermost-load-test-ng/blob/7e0ab74f961548e432907ea7803d0560aad70b32/loadtest/control/simplecontroller/actions.go#L116). That was my main point actually.

Reply:
Hmm .. that's true actually. I think we should take out the permission checking from inside the action then. Let me update.